optimum/amd/brevitas/export.py

```diff
@@ -1,15 +1,256 @@
+            [f"matmul_integer_{i}"],
+        )
+        compute_graph = create_nodes(
+            compute_graph, "Cast", f"cast_{i}", [f"matmul_integer_{i}"], [f"cast_{i}"], to=int(1)
+        )
+        compute_graph = create_nodes(
+            compute_graph, "Mul", f"mulscales_{i}", [dyn_q.output[1], f"dq_weights_1_{i}"], [f"mulscales_{i}"]
+        )
+        compute_graph = create_nodes(
+            compute_graph, "Mul", f"mulvalues_{i}", [f"mulscales_{i}", f"cast_{i}"], [matmul.output[0]]
+        )
+    return compute_graph
+
+
+def replace_gemm_to_matmulinteger(compute_graph, found_nodes):
+    k = 100
+    for i, found_pattern in enumerate(found_nodes):
+        k = i + 100
+        gemm = compute_graph.nodemap[found_pattern[-1]]
+        bias = gemm.input[-1]
+        deq_linear = compute_graph.nodemap[found_pattern[0]]
+        dyn_q = compute_graph.nodemap[found_pattern[1]]
+        dq_weight = deq_linear.prevnodes[0]
+        compute_graph.add_initial(f"dq_weights_0_{k}", dq_weight.value.transpose())
+        compute_graph.add_initial(f"dq_weights_1_{k}", deq_linear.prevnodes[1].value)
+        compute_graph.add_initial(f"dq_weights_2_{k}", deq_linear.prevnodes[2].value)
+
+        matmul = compute_graph.nodemap[found_pattern[-1]]
+        for name in found_pattern:
+            if "DynamicQuantizeLinear" in name:
+                continue
+            compute_graph.remove_node(name)
+        compute_graph.remove_node(deq_linear.prevnodes[0].name)
+        compute_graph.remove_node(deq_linear.prevnodes[1].name)
+        if deq_linear.prevnodes[2].name in compute_graph.nodemap:
+            compute_graph.remove_node(deq_linear.prevnodes[2].name)
+
+        compute_graph = create_nodes(
+            compute_graph,
+            "MatMulInteger",
+            f"matmul_integer_{k}",
+            [dyn_q.output[0], f"dq_weights_0_{k}", dyn_q.output[2], f"dq_weights_2_{k}"],
+            [f"matmul_integer_{k}"],
+        )
+        compute_graph = create_nodes(
+            compute_graph, "Cast", f"cast_{k}", [f"matmul_integer_{k}"], [f"cast_{k}"], to=int(1)
+        )
+        compute_graph = create_nodes(
+            compute_graph, "Mul", f"mulscales_{k}", [dyn_q.output[1], f"dq_weights_1_{k}"], [f"mulscales_{k}"]
+        )
+        compute_graph = create_nodes(
+            compute_graph, "Mul", f"mulvalues_{k}", [f"mulscales_{k}", f"cast_{k}"], [f"mulvalues_{k}"]
+        )
+        compute_graph = create_nodes(
+            compute_graph, "Add", f"addbias_{k}", [bias, f"mulvalues_{k}"], [matmul.output[0]]
+        )
+    return compute_graph
+
+
+def find_and_insert_matmulinteger(model_path):
+    print("Rewriting ONNX Graph with MatMulInteger ")
+    model_path = os.path.join(model_path, "model.onnx")
+    cfg = {"constant_folding": False, "node_rename": False, "if_fixed_branch": None, "fixed_topk": 0, "verbose": True}
+    original_output = onnx.load(model_path).graph.output
+    model = Model(model_path, cfg)
```

Comment:
I am not familiar with onnx_tool `Model`, but isn't the model loaded twice here (doubling load time)? Can't we initialize a `Model` from a `ModelProto`, or alternatively retrieve `original_output` from the `Model`?

Reply:
Yeah you're right, it should work